source/agora/common/Task.d

```diff
@@ -73,14 +73,57 @@ public class TaskManager
             dg = This delegate will be called when the timer fires.
             periodic = Specifies if the timer fires repeatedly or only once.
 
+        Returns:
+            An `ITimer` interface with the ability to control the timer
+
     ***************************************************************************/
 
-    public void setTimer (Duration timeout, void delegate() dg,
+    public ITimer setTimer (Duration timeout, void delegate() dg,
         Periodic periodic = Periodic.No)
     {
         assert(dg !is null, "Cannot call this delegate if null");
-
         static import vibe.core.core;
-        vibe.core.core.setTimer(timeout, dg, periodic);
+        return new VibedTimer(vibe.core.core.setTimer(timeout, dg, periodic));
+    }
+}
+
+/*******************************************************************************
+
+    Defines an abstraction over the timer implementation
+
+*******************************************************************************/
+
+public interface ITimer
+{
+    /***************************************************************************
+
+        Stop or cancel this timer
+
+    ***************************************************************************/
+
+    public void stop ();
+}
+
+/*******************************************************************************
+
+    Vibe.d timer
+
+*******************************************************************************/
+
+private final class VibedTimer : ITimer
+{
+    static import Vibe = vibe.core.core;
+
+    private Vibe.Timer timer;
+
+    public this (Vibe.Timer timer) @safe nothrow
```

Comment:
Missing `pure @nogc`

Reply:
`source/agora/common/Task.d(119,12): Error: pure constructor agora.common.Task.VibedTimer.this cannot call impure destructor vibe.core.core.Timer.~this
source/agora/common/Task.d(119,12): Error: @nogc constructor agora.common.Task.VibedTimer.this cannot call non-@nogc destructor vibe.core.core.Timer.~this
source/agora/common/Task.d(121,20): Error: pure constructor agora.common.Task.VibedTimer.this cannot call impure function vibe.core.core.Timer.__postblit
source/agora/common/Task.d(121,20): Error: @nogc constructor agora.common.Task.VibedTimer.this cannot call non-@nogc function vibe.core.core.Timer.__postblit`

Reply:
Oh man. LGTM then.